opentracing/tracecontext.go

```
@@ -49,18 +49,18 @@ type TraceContext interface {
 	// HTTP headers are case insensitive.
 	//
 	// `restrictedKey` MUST match the regular expression
-	// `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. See
-	// CanonicalizeTraceTagKey().
+	// `(?i:[a-z0-9][-a-z0-9]*)` and is case-insensitive. That is, it must
+	// start with a letter or number, and the remaining characters must be
+	// letters, numbers, or hyphens. See CanonicalizeTraceTagKey().
 	//
 	// Returns a reference to this TraceContext for chaining, etc.
-	SetTraceTag(caseInsensitiveKey, value string) TraceContext
+	SetTraceTag(restrictedKey, value string) TraceContext
```

Much better, thank you @bensigelman :+1: 

I still think we should mention that implementations should panic on invalid keys, though?